src/main/kotlin/org/arend/intention/RemoveClarifyingParensIntention.kt

```diff
@@ -0,0 +1,152 @@
+}
+
+private fun parentArgumentExpression(tuple: ArendTuple): PsiElement? {
+    return tuple.parentOfType<ArendAtomFieldsAcc>()?.let {
+        if (it.parent is ArendAtomArgument) it.parent else it
+    }
+}
+
+private fun hasClarifyingParens(binOpSeq: Concrete.AppExpression): Boolean {
+    if ((binOpSeq.data as? PsiElement)?.textContains('(') == false) {
+        return false
+    }
+    var clarifyingParensFound = false
+    val queue = mutableListOf(binOpSeq)
+    while (!clarifyingParensFound && queue.isNotEmpty()) {
+        val parentBinOpApp = queue.removeFirst()
+        for (arg in parentBinOpApp.arguments) {
+            if (!arg.isExplicit) {
+                continue
+            }
+            val expression = arg.expression
+            if (expression is Concrete.HoleExpression) {
+                val binOp = findBinOpInParens(expression)
+                if (binOp != null) {
+                    if (doesNotNeedParens(binOp, parentBinOpApp)) {
+                        clarifyingParensFound = true
+                        break
+                    }
+                    queue.add(binOp)
+                }
+            }
+            if (expression is Concrete.AppExpression && BinOpIntentionUtil.isBinOpApp(expression)) {
+                queue.add(expression)
+            }
+        }
+    }
+    return clarifyingParensFound
+}
+
+private fun findBinOpInParens(expression: Concrete.HoleExpression): Concrete.AppExpression? {
+    val tuple = (expression.data as? ArendAtomFieldsAcc)?.childOfType<ArendTuple>() ?: return null
+    val appExprPsi = unwrapAppExprInParens(tuple) ?: return null
+    return BinOpIntentionUtil.toConcreteBinOpApp(appExprPsi)
+}
+
+private fun unwrapAppExprInParens(tuple: ArendTuple): ArendArgumentAppExpr? {
+    val expr = unwrapParens(tuple) ?: return null
+    return (expr as? ArendNewExpr)?.appExpr as? ArendArgumentAppExpr ?: return null
+}
+
+private fun unwrapParens(tuple: ArendTuple): ArendExpr? {
+    val tupleExpr = tuple.tupleExprList.singleOrNull() ?: return null
+    return if (tupleExpr.colon == null) tupleExpr.exprList.singleOrNull() else null
+}
+
+private fun doesNotNeedParens(childBinOp: Concrete.AppExpression, parentBinOp: Concrete.AppExpression): Boolean {
+    val childPrecedence = getPrecedence(childBinOp.function) ?: return false
+    val parentPrecedence = getPrecedence(parentBinOp.function) ?: return false
+    val childIsLeftArg = rangeOfConcrete(childBinOp).endOffset < rangeOfConcrete(parentBinOp.function).startOffset
+    return if (childIsLeftArg)
+        comparePrecedence(childPrecedence, parentPrecedence) == PartialComparator.Result.GREATER
+    else comparePrecedence(parentPrecedence, childPrecedence) == PartialComparator.Result.LESS
+}
+
+private fun getPrecedence(function: Concrete.Expression) =
+        ((function as? Concrete.ReferenceExpression)?.referent as? GlobalReferable)?.precedence
+
+// TODO copy-pasted from MetaBinOpParser
```

Comment:
`MetaBinOpParser#comparePrecedence` is private, so I have copy-pasted it here. Is it OK if I make it public in the Arend repo and drop this copy?